buildSrc/src/main/kotlin/org/jetbrains/conventions/kotlin-jvm.gradle.kts

```diff
@@ -0,0 +1,47 @@
+package org.jetbrains.conventions
+
+import org.jetbrains.configureDokkaVersion
+import org.jetbrains.isLocalPublication
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
+plugins {
+    id("org.jetbrains.conventions.base")
+    kotlin("jvm")
+    id("org.jetbrains.dokka")
+}
+
+configureDokkaVersion()
+
+tasks.dokkaHtml {
+    onlyIf { !isLocalPublication }
+    outputDirectory.set(layout.buildDirectory.dir("dokka").map { it.asFile })
+}
+
+val javadocJar by tasks.registering(Jar::class) {
+    archiveClassifier.set("javadoc")
+    from(tasks.dokkaHtml)
+}
```

Comment:
Good question! I think explaining this is a demo of Gradle concepts.

Moving this task to the `maven-publish` convention is a little bit tricky because that convention plugin doesn't have the Dokka plugin applied, so `tasks.dokkaHtml` might not exist. Even if a subproject did apply both Dokka and `maven-publish`, if it applies them in the wrong order, the task still might not exist. (This is why the Gradle Kotlin DSL is great - it will protect you against using missing tasks by only generating the accessor when the task is present.)

You _could_ move this task to the `maven-publish` convention but wrap it in a block that will only register the task when the Dokka plugin is present

```kotlin
plugins.withId("org.jetbrains.dokka") {
    // define javadocJar task...
}
```

(or an equivalent 'plugin reaction config' in the `kotlin-jvm` that will only define the javadocJar task if `maven-publish` is present)

But... both of these aren't really necessary. It's not really a problem if a subproject has a task that it doesn't need, because the task will only be run when it's required. And the javadocJar task is only required when a publication is triggered.

https://github.com/Kotlin/dokka/blob/8c0344e682fc4ba0e44fc740269c7ba308c47c35/buildSrc/src/main/kotlin/org/jetbrains/publication.kt#L36